### PR TITLE
fix: login and signup component unit tests

### PR DESCRIPTION
**Pull request by kgajera**
This fixes the login and signup component unit tests.

## Changes

The tests were failing because:
- The `prefetch` function on our mock router needed to return a promise
- The `@testing-library` dependencies required updating

## Checklist

- [x] Requires dependency update?
- [x] Generating a new app works

### CODE REVIEW

File: packages/create-bison-app/template/tests/utils.tsx [hunk 1]
```diff
@@ -80,8 +80,8 @@ const mockRouter: NextRouter = {
   replace: jest.fn(),
   reload: jest.fn(),
   back: jest.fn(),
-  prefetch: jest.fn(),
   beforePopState: jest.fn(),
+  prefetch: jest.fn(() => Promise.resolve()),
```
Comment (cmejet) [inline packages/create-bison-app/template/tests/utils.tsx:84]: 🎉

**Review by cmejet — CHANGES_REQUESTED** [review]
This is awesome!
Given the other PR was merged that skipped those tests we will want to make sure we rebase from canary before we merge this but can't wait to have all the tests working again.

**Comment by kgajera** [comment]
I just rebased the PR and updated the tests to not be skipped.

Screen Shot 2021-06-18 at 9 40 35 AM

**Review by cmejet — APPROVED** [review]
image
🎉

**Pull request merged**